http-server/src/main/java/io/micronaut/http/server/exceptions/response/BodyErrorResponseProvider.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.server.exceptions.response;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.http.HttpResponse;
+
+/**
+ * Provides an HTTP Response body of an error response.
+ * @author Sergio del Amo
+ * @since 4.7.0
+ *  @param <T> The body type
+ */
+public interface BodyErrorResponseProvider<T> {
```

Comment:
rename to `ErrorResponseBodyProvider` which is more logical a name

Reply:
https://github.com/micronaut-projects/micronaut-core/pull/11210/commits/a36464e806064b6123641db2699fdede29b0d1e9